internal/pkg/template/templates/workloads/partials/cf/alb.yml

```diff
@@ -23,10 +23,14 @@ TargetGroup:
     {{- if .HTTPHealthCheck.Timeout}}
     HealthCheckTimeoutSeconds: {{.HTTPHealthCheck.Timeout}}
     {{- end}}
-    {{- if .HTTPTargetContainer.IsHTTPS }}
+    {{/* if health check port is configured to 443, or if health check port not specified and target container is https */}}
+    {{- if or (and .HTTPHealthCheck.Port .HTTPHealthCheck.IsHTTPS) (and (not .HTTPHealthCheck.Port) .HTTPTargetContainer.IsHTTPS)}}
```

Comment:
Would it be better (more readable) if we move this logic to the function itself 🤔 Like can we combine

```
	HTTPTargetContainer     HTTPTargetContainer
	HTTPHealthCheck         HTTPHealthCheckOpts
```

to 

```
type HTTPTarget struct {
    HTTPTargetContainer
    HTTPHealthCheckOpts
}
```

Reply:
👍 or we can add a template function too, something like:
```yaml
HealthCheckProtocol: {{healthCheckProtocol .HTTPHealthCheck .HTTPTargetContainer}}
```
That returns either "HTTP" or "HTTPS"